src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -1378,6 +1403,11 @@ private void RemoveLastInputPrompt()
             private void AddLanguageBuffer()
             {
                 ITextBuffer buffer = _factory.CreateAndActivateBuffer(_window);
+                if (CurrentLanguageBuffer != null)
+                {
+                    _textBufferUndoManagerProvider.RemoveTextBufferUndoManager(CurrentLanguageBuffer);
+                }
+                UndoManager = _textBufferUndoManagerProvider.GetTextBufferUndoManager(buffer);
```

Comment:
I actually tried that first, but undo doesn't work since the surface buffer (`TextView.TextBuffer`) includes readonly buffers.